From 53e384d2c0f516f4cf7cc2051bf3c3c38996e1d5 Mon Sep 17 00:00:00 2001
From: Heinrich Schuchardt <xypron.glpk@gmx.de>
Date: Wed, 29 Jul 2020 12:49:24 +0200
Subject: [PATCH] riscv: load addresses for Sipeed MAIX

Define default load addresses and the device tree name for the Sipeed MAIX.

Signed-off-by: Heinrich Schuchardt <xypron.glpk@gmx.de>
---
v2:
	leave 128 KiB for OpenSBI at 0x80000000
---
 include/configs/sipeed-maix.h | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/include/configs/sipeed-maix.h b/include/configs/sipeed-maix.h
index a46473fc78..978db6fb00 100644
--- a/include/configs/sipeed-maix.h
+++ b/include/configs/sipeed-maix.h
@@ -21,4 +21,13 @@
 /* For early init */
 #define K210_SYSCTL_BASE 0x50440000
 
+#ifndef CONFIG_EXTRA_ENV_SETTINGS
+#define CONFIG_EXTRA_ENV_SETTINGS \
+	"loadaddr=0x80060000\0" \
+	"fdt_addr_r=0x80028000\0" \
+	"scriptaddr=0x80020000\0" \
+	"kernel_addr_r=0x80060000\0" \
+	"fdtfile=kendryte/" CONFIG_DEFAULT_DEVICE_TREE ".dtb\0"
+#endif
+
 #endif /* CONFIGS_SIPEED_MAIX_H */
-- 
2.27.0

